homeassistant/components/noaa_tides/sensor.py

```diff
@@ -49,41 +60,138 @@
+            f"deprecated_yaml_{DOMAIN}",
+            breaks_in_ha_version=DEPRECATE_YAML_IN_HA_VERSION,
+            is_fixable=False,
+            issue_domain=DOMAIN,
+            severity=IssueSeverity.WARNING,
+            translation_key="deprecated_yaml",
+            translation_placeholders={
+                "domain": DOMAIN,
+                "integration_title": NAME,
+            },
+        )
         return
+
+    if (
+        result["type"] is FlowResultType.FORM
+        and result["errors"] is not None
+        and result["errors"]["base"] is not None
+    ):
+        _LOGGER.error(
+            "Cannot import %s from configuration.yaml (station_id = %s): %s",
+            DOMAIN,
+            station_id,
+            result["errors"]["base"],
+        )
+        async_create_issue(
+            hass,
+            DOMAIN,
+            f"deprecated_yaml_import_issue_{result['errors']['base']}",
+            breaks_in_ha_version=DEPRECATE_YAML_IN_HA_VERSION,
+            is_fixable=False,
+            issue_domain=DOMAIN,
+            severity=IssueSeverity.WARNING,
+            translation_key=f"deprecated_yaml_import_issue_{result['errors']['base']}",
+            translation_placeholders={
+                "domain": DOMAIN,
+                "integration_title": NAME,
+            },
+        )
+        return
+
+    async_create_issue(
+        hass,
+        DOMAIN,
+        "deprecated_yaml_import_issue_unknown",
+        breaks_in_ha_version="2025.2.0",
+        is_fixable=False,
+        issue_domain=DOMAIN,
+        severity=IssueSeverity.WARNING,
+        translation_key="deprecated_yaml_import_issue_unknown",
+        translation_placeholders={
+            "domain": DOMAIN,
+            "integration_title": NAME,
+        },
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Add NOAA Tides sensor entry."""
+
+    station_id = entry.data.get(CONF_STATION_ID)
+    name = entry.title
+    timezone = entry.options.get(CONF_TIME_ZONE, DEFAULT_TIMEZONE)
+    unit_system = entry.options.get(CONF_UNIT_SYSTEM, get_default_unit_system(hass))
+
+    if station_id is None:
+        _LOGGER.error("Station ID is required")
+        raise StationNotFound
+
+    try:
+        station = await hass.async_add_executor_job(
+            coops.Station, station_id, unit_system
+        )
+    except KeyError as exception:
+        _LOGGER.error("%s sensor station_id %s does not exist", NAME, station_id)
+        raise StationNotFound from exception
     except requests.exceptions.ConnectionError as exception:
         _LOGGER.error(
-            "Connection error during setup in NOAA Tides Sensor for station_id: %s",
+            "Connection error during setup in %s sensor for station_id: %s",
+            NAME,
             station_id,
         )
         raise PlatformNotReady from exception
```

Comment:
Let's have this logic in `__init__.py` and then just store it in `entry.runtime_data`

Reply:
Let me know if I did this correctly. This is my first time using runtime_data.